Frameworks/Foundation/_UTTypeInternal.mm

```diff
@@ -0,0 +1,585 @@
+            }
+
+            // Add each file name extension to the s_fileExtensionToSystemUTIMap
+            //  Key: File Name Extension (CFStringRef)
+            //  Value: CFMutableArray of UTI strings (CFMutableArrayRef)
+            for (const auto& fileNameExtension : entry.fileNameExtensions) {
+                if (!CFDictionaryContainsKey(s_fileExtensionToSystemUTIMap, (const void*)fileNameExtension)) {
+                    CFMutableArrayRef systemUTIArray = CFArrayCreateMutable(kCFAllocatorDefault, 0, &kCFTypeArrayCallBacks);
+                    CFArrayAppendValue(systemUTIArray, (const void*)entry.utiName);
+                    CFDictionaryAddValue(s_fileExtensionToSystemUTIMap, (const void*)fileNameExtension, (const void*)systemUTIArray);
+                } else {
+                    CFMutableArrayRef systemUTIArray = static_cast<CFMutableArrayRef>(
+                        const_cast<void*>(CFDictionaryGetValue(s_fileExtensionToSystemUTIMap, (const void*)fileNameExtension)));
+                    CFArrayAppendValue(systemUTIArray, (const void*)entry.utiName);
+                    CFDictionarySetValue(s_fileExtensionToSystemUTIMap, (const void*)fileNameExtension, (const void*)systemUTIArray);
+                }
+            }
+
+            // Add each mime type to the s_mimeTypeToSystemUTIMap
+            //  Key: Mime Type (CFStringRef)
+            //  Value: CFMutableArray of UTI strings (CFMutableArrayRef)
+            for (const auto& mimeType : entry.mimeTypes) {
+                if (!CFDictionaryContainsKey(s_mimeTypeToSystemUTIMap, (const void*)mimeType)) {
+                    CFMutableArrayRef systemUTIArray = CFArrayCreateMutable(kCFAllocatorDefault, 0, &kCFTypeArrayCallBacks);
+                    CFArrayAppendValue(systemUTIArray, (const void*)entry.utiName);
+                    CFDictionaryAddValue(s_mimeTypeToSystemUTIMap, (const void*)mimeType, (const void*)systemUTIArray);
+                } else {
+                    CFMutableArrayRef systemUTIArray = static_cast<CFMutableArrayRef>(
+                        const_cast<void*>(CFDictionaryGetValue(s_mimeTypeToSystemUTIMap, (const void*)mimeType)));
+                    CFArrayAppendValue(systemUTIArray, (const void*)entry.utiName);
+                    CFDictionarySetValue(s_mimeTypeToSystemUTIMap, (const void*)mimeType, (const void*)systemUTIArray);
+                }
+            }
+        }
+    });
+}
+
+/**
+ * Retrieves the UTI for the given file name extension.
+ * @param fileNameExtension file name extension
+ * @return UTI
+ */
+CFArrayRef _UTGetUTIsForFileNameExtension(CFStringRef fileNameExtension) {
+    if (fileNameExtension == NULL) {
+        return NULL;
+    }
+
+    CFArrayRef UTIs =
+        static_cast<CFArrayRef>(const_cast<void*>(CFDictionaryGetValue(s_fileExtensionToSystemUTIMap, (const void*)fileNameExtension)));
+    return UTIs ? CFArrayCreateCopy(kCFAllocatorDefault, UTIs) : NULL;
```

Comment:
this is old code that was moved, clearly this is confusing, names need an update.

---
In reply to: [95278794](https://github.com/Microsoft/WinObjC/pull/1662#discussion_r95278794) [](ancestors = 95278794)